OpenRA.Mods.Common/SpriteLoaders/GifLoader.cs

```diff
@@ -0,0 +1,308 @@
+﻿using OpenRA.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenRA.Mods.Common.SpriteLoaders
+{
+	class GifLoader : ISpriteLoader
+	{
+		public class GifFrame : ISpriteFrame
+		{
+			public Size Size { get; private set; }
+			public Size FrameSize { get; private set; }
+			public float2 Offset { get { return offset; } }
+
+			public byte[] Data { get; set; }
+			public bool DisableExportPadding { get { return false; } }
+
+			float2 offset;
+
+			public GifFrame(Stream s, Size size)
+			{
+				var imageLeft = s.ReadUInt16();
+				var imageTop = s.ReadUInt16();
+				var imageWidth = s.ReadUInt16();
+				var imageHeight = s.ReadUInt16();
+
+				var unpacked = s.ReadUInt8();
+
+				// Local palette
+				if (GetBitValue(unpacked, 0) != 0)
+				{
+					var bits = new byte[]{
+						GetBitValue(unpacked, 5),
+						GetBitValue(unpacked, 6),
+						GetBitValue(unpacked, 7)
+					};
+					
+					s.Position += 3 * (1 << (BitsToNum(bits) + 1));
```

Comment:
This looks like an extremely round-about way of doing... whatever it is you're doing.